docs/prompting/thought_generation/chain_of_thought_few_shot/active_prompt.md

```diff
@@ -1,7 +1,99 @@
 ---
-title: ""
+title: "Active Prompting"
 description: ""
-keywords: ""
 ---
 
-[wip]
+# Active Prompting
+
+Active prompting<sup><a href="https://arxiv.org/abs/2302.12246">1</a></sup> is a method used to identify the most effective examples for human annotation. It is particularly useful when dealing with large amounts of unlabeled data, as it provides a system to prioritze which data to label by humans. The process involves four key steps:
+
+1. **Uncertainty Estimation**: Assessing the uncertainty of the model's predictions
+2. **Selection**: Choosing the most uncertain examples for human annotation
+3. **Annotation**: Having humans label the selected examples
+4. **Inference**: Using the newly labeled data to improve the model's performance
+
+## Uncertainty Estimation
+
+In this step, we define an unsupervised method to measure the uncertainty of an LLM in answering a given example.
+
+!!! example "Uncertainty Estimation Example"
+
+    Let's say we ask an LLM the following query:
+    >query = "Classify the sentiment of this sentence as positive or negative: I am very excited today."
+
+    and the LLM returns:
+    >response = "positive"
+
+    The goal of uncertainty estimation is to answer: **How sure is the LLM in this response?**
+
+In order to do this, we query the LLM with the same example _k_ times. Then, we use the _k_ responses to determine how dissimmilar these responses are. Three possible metrics<sup><a href="https://arxiv.org/abs/2302.12246">1</a></sup> are:
+
+1. Disagreement
+2. Entropy
+3. Variance
+
+Below is an example of uncertainty estimation for a single input example using the disagreement uncertainty metric.
+
+```python
+import instructor
+from pydantic import BaseModel
+from openai import OpenAI
+
+class UserInfo(BaseModel):
+    name: str
+    age: int
+
+client = instructor.from_openai(OpenAI())
+
+k = 5 # (1)!
+responses = []
+
+# Query the LLM k times
+for _ in range(k):
+    user_info = client.chat.completions.create(
+        model="gpt-4o",
+        response_model=UserInfo,
+        messages=[{"role": "user", "content": "How old is Jason Liu?"}],
+    )
```

Comment:
use a better example 

even 

results = [ function() for _ in range(10) ]

vs all this append code